### ISSUE TITLE
Should _isCursor check for a cursor on props.cursor?

### ISSUE THREAD

**Issue by phated** [omniscientjs/omniscient#52]
It looks to me like the way omniscient worked before refactor is that it took a cursor given as props and attached it to `props.cursor`, but now it takes a cursor as `props` directly, attaches it to a hidden property and then uses it inside the render method.  Would it make sense to take a cursor that was passed in as `props.cursor` and do the same thing as if the cursor is passed directly as `props`?  This would make it easier to work with statics and cursors.

**Comment by Dashed** [comment]
In what way would make it easier with statics and cursors?

**Comment by mikaelbr** [comment]
Hi @phated and sorry for the late response!

This is somewhat related to #43 (see https://github.com/omniscientjs/omniscient/issues/43#issuecomment-72929035). So I think this would be surprising to the user, where as if you send something in in a JSX component like `<Component cursor={cursor} />`, and you are use to that this means accessing through `props.cursor` or `this.props.cursor`, it could feel inconsistent. 

How would this translate?
```jsx
<Component cursor={cursor} someOther="foo" />
```

`someOther` here is not a part of statics, and changing it would cause a re-render, so you would have to have it as a part of props, so unboxing just cursor and passing that on as input wouldn't make sense, IMO.

In this however, `someOther` would be a part of `statics`

```jsx
<Component cursor={cursor} statics={{someOther: "foo"}} />
```

But would it make sense to do unbox the cursor, ignoring the statics?
I don't think "magical" unboxing is the way to go.

On a related note, https://github.com/omniscientjs/omniscient/commit/a523e595a0b34807ce6b0d068287362267bfc476#diff-85be81e3538eeba3d115e6ea83f12809R223, might do it easier to use statics for non-JSX usage.

**Comment by phated** [comment]
It seems like you hashed a lot of this out in #43 

My thinking was that the component function signature would be:

```jsx
var Component = component(function(cursor, statics){
  return <h1>{cursor.get('title')}</h1>
});
```
and could pull the `cursor` property from `this.props` if it exists.
```jsx
<Component cursor={cursor} />
```
Since you still allow consumers to access `this.props`, any non-cursor/non-statics can still be accessed and handled on a re-render.

This would impose a convention which I'm not sure you are comfortable with and it might make things more confusing when you pass in `this.props` if `this.props.cursor` is undefined.

That being said, since I opened this issue, I've started using destructuring for the props and it makes it almost as nice as my proposed API.  So feel free to close this if you are happy with the destructuring solution.
